### PR TITLE
Adds the TAsyncPusherServer to allow async requests

### PR DESCRIPTION
**Pull request by RobertoSchneiders**

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -55,6 +55,22 @@ SSL is enabled by default. You can disable it by passing a empty option list `[]
 TPusherServer.Create('app_id', 'app_key', 'secret', '', []);
 ```
 
+### Async
+
+To use async request you should use the `TAsyncPusherServer` class:
+```
+AsyncPusherServer := TAsyncPusherServer.Create('app_id', 'app_key', 'secret');
```
Comment (fabioxgn) [inline README.md:62]: I don't think this will be usefull without the `OnError` event handler, why not add the OnError as a parameter in the constructor?

**Comment by martinusso** [comment]
:+1:

**Comment by fabioxgn** [comment]
:+1:  I would add the OnError in the constructor, so it is harder to miss implementing it.

File: README.md (hunk 1 above)
Comment (RobertoSchneiders) [inline README.md:62]: Makes sense, but, I don't know if it is a good idea.

Indeed, it probably doesn't make sense to use this class without this event, but I'm not sure it is worth to add it to the constructor. 

I don't like the syntax for passing an anonymous method to a constructor which already have other parameters.

Comment (fabioxgn) [inline README.md:62]: @RobertoSchneiders :+1:  with an anonymous method it would look ugly indeed.

**Pull request merged**